src/sqlfluff/core/linter/linting_result.py

```diff
@@ -181,3 +184,34 @@ def tree(self) -> Optional[BaseSegment]:  # pragma: no cover
                 "path."
             )
         return self.paths[0].tree
+
+    def check_templating_or_parse_errors(self) -> Tuple[int, int]:
+        """Treat tiles with templating or parse error as unfixable.
+
+        NOTE! THIS FUNCTION HAS SIDE EFFECTS: IT DISCARDS LINT FIXES.
+
```

Comment:
I can see why this function has side effects. Ideally it wouldn't, but I think they way this function is used it's more of a pruning function or filter function on the current set of violations.

Given that it *does* have side effects however, I wonder if we should change the name. `check_<x>` makes it sound like a function which _doesn't_ have side effects.

Reply:
Done.